src/interface/src/app/signup/signup.component.ts

```diff
@@ -52,25 +53,36 @@ export class SignupComponent {
   signup() {
     if (this.submitted) return;
 
+    this.submitted = true;
+
     const email: string = this.form.get('email')?.value;
     const password1: string = this.form.get('password1')?.value;
     const password2: string = this.form.get('password2')?.value;
     const firstName: string = this.form.get('firstName')?.value;
     const lastName: string = this.form.get('lastName')?.value;
     this.authService
       .signup(email, password1, password2, firstName, lastName)
+      .pipe(timeout(10000))
```

Comment:
fyi, in case is useful on your error code, this will throw a `TimeoutError`, and additionally, you can provide a custom error https://rxjs.dev/api/operators/timeout